src/ast/rewriter/seq_rewriter.cpp

```diff
@@ -2450,40 +2484,91 @@ expr_ref seq_rewriter::mk_derivative_rec(expr* ele, expr* r) {
         }
         expr_ref dr2 = mk_derivative(ele, r2);
         is_n = re_predicate(is_n, seq_sort);
-        return mk_der_union(result, mk_der_concat(is_n, dr2));        
+        if (lift_over_union) {
+            return mk_der_union(result, mk_der_concat(is_n, dr2));
+        }
+        else {
+            return expr_ref(re().mk_union(result, mk_der_concat(is_n, dr2)), m());
+        }
+    }
+    else if (re().is_concat(r, r1, r2) && !left) {
+        expr_ref is_n = is_nullable(r2);
+        expr_ref dr2 = mk_derivative(ele, r2, left);
+        result = mk_der_concat(r1, dr2);
+        if (m().is_false(is_n)) {
+            return result;
+        }
+        expr_ref dr1 = mk_derivative(ele, r1, left);
+        is_n = re_predicate(is_n, seq_sort);
+        if (lift_over_union) {
+            return mk_der_union(result, mk_der_concat(dr1, is_n));
```

Comment:
dr2?

Reply:
This is the right-hand case so it should be dr1, but the error is actually in line 2507 (that should be dr1)